packages/fxa-auth-server/test/scripts/stripe-products-and-plans-converter.js

```diff
@@ -0,0 +1,798 @@
+      Container.set(PaymentConfigManager, paymentConfigManager);
+      productConfigDbRef = paymentConfigManager.productConfigDbRef;
+      planConfigDbRef = paymentConfigManager.planConfigDbRef;
+      converter = new StripeProductsAndPlansConverter({
+        log: mockLog,
+        stripeHelper: mockStripeHelper,
+        supportedLanguages: mockSupportedLanguages,
+      });
+      args = {
+        productId: '',
+        isDryRun: false,
+      };
+      async function* productGenerator() {
+        yield product1;
+        yield product2;
+      }
+      async function* planGenerator1() {
+        yield plan1;
+        yield plan2;
+      }
+      async function* planGenerator2() {
+        yield plan3;
+      }
+      converter.stripeHelper.stripe = {
+        products: { list: sandbox.stub().returns(productGenerator()) },
+        plans: {
+          list: sandbox
+            .stub()
+            .onFirstCall()
+            .returns(planGenerator1())
+            .onSecondCall()
+            .returns(planGenerator2()),
+        },
+      };
+    });
+    afterEach(async () => {
+      await deleteCollection(
+        paymentConfigManager.firestore,
+        productConfigDbRef,
+        100
+      );
+      await deleteCollection(
+        paymentConfigManager.firestore,
+        planConfigDbRef,
+        100
+      );
+      Container.reset();
+      sandbox.reset();
+    });
+    it('processes new products and plans', async () => {
+      await converter.convert(args);
+      await retry(
+        async () => {
+          products = paymentConfigManager.allProducts();
+          plans = paymentConfigManager.allPlans();
+
+          assert.equal(products.length, 2);
+          assert.equal(plans.length, 3);
+        },
+        {
+          retries: 10,
+          minTimeout: 20,
+        }
+      );
+      // We don't care what the values of the Firestore doc IDs as long
+      // as they match the expected productConfigId for planConfigs.
+      assert.deepEqual(products[0], {
+        ...productConfig1,
+        id: products[0].id,
+      });
+      assert.deepEqual(products[1], {
+        ...productConfig2,
+        id: products[1].id,
+      });
+      assert.deepEqual(plans[0], {
+        ...planConfig1,
+        id: plans[0].id,
+        productConfigId: products[0].id,
+      });
+      assert.deepEqual(plans[1], {
+        ...planConfig2,
+        id: plans[1].id,
+        productConfigId: products[0].id,
+      });
+      assert.deepEqual(plans[2], {
+        ...planConfig3,
+        id: plans[2].id,
+        productConfigId: products[1].id,
+      });
+    });
+    it('updates existing products and plans', async () => {
+      // Put some configs into Firestore
+      const productConfigDocId1 = await paymentConfigManager.storeProductConfig(
+        productConfig1
+      );
+      await paymentConfigManager.storePlanConfig(
+        planConfig1,
+        productConfigDocId1
+      );
```

Comment:
@bbangert I was previously calling `await converter.convert(args);` here which was calling `paymentConfigManager.load`, but as we started discussing Friday, there seems to be a bug when `load` is called and items are already in the Firestore emulator.

I did some more debugging, and I narrowed it down to `ProductConfig.fromFirestoreObject` and `PlanConfig.fromFirestoreObject`, but I didn't get much further. Hoping we can pair more on this today. I'm happy to file a bug to investigate this too to avoid further scope creep here.

I also had to work around this bug in my new `PaymentConfigManager.getDocumentIdByStripeId` helper. I mocked `allProducts` and `allPlans` instead of calling `load`.

Reply:
I filed https://mozilla-hub.atlassian.net/browse/FXA-4706 to cover this.

Reply:
Ah, sorry, I thought you and Ben already paired on this since the comment's from six days ago.  😬